soc/st/stm32/stm32n6x/soc.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for STM32N6 processor
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/cache.h>
+#include <zephyr/logging/log.h>
+
+#include <stm32_ll_bus.h>
+#include <stm32_ll_pwr.h>
+#include <stm32_ll_icache.h>
+
+#include <cmsis_core.h>
+
+#define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
+LOG_MODULE_REGISTER(soc);
+
+extern char _vector_start[];
+void *g_pfnVectors = (void *)_vector_start;
+
+#if defined(CONFIG_SOC_RESET_HOOK)
+void soc_reset_hook(void)
+{
+	/* This is provided by STM32Cube HAL */
+	SystemInit();
```

Comment:
Can we replace directly by the following to minimize dependencies ?
```
  SCB->VTOR = INTVECT_START;
```
(As I guess `USER_VECT_TAB_ADDRESS` is not defined).

And btw, isn't it already set as part of zephr arch init ?

Reply:
SystemInit contains a lot more initialization than just this line :)

Reply:
ok, indeed, my bad. Let's keep it that way then.